lib/active_merchant/billing/gateways/globalcollect.rb

```diff
@@ -0,0 +1,294 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class GlobalcollectGateway < Gateway
+      self.display_name = "GlobalCollect"
+      self.homepage_url = "http://www.globalcollect.com/"
+
+      self.test_url = "https://api-sandbox.globalcollect.com/"
+      self.live_url = "https://api.globalcollect.com/"
+
+      self.supported_countries = %w(AD AE AT AU BD BE BG BN CA CH CY CZ DE DK
+      EE EG ES FI FR GB GI GR HK HU ID IE IL IM IN IS IT JO KW LB LI LK LT LU
+      LV MC MT MU MV MX MY NL NO NZ OM PH PL PT QA RO SA SE SG SI SK SM TR TT
+      UM US VA VN ZA)
+      self.default_currency = "USD"
+      self.money_format = :cents
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      def initialize(options={})
+        requires!(options, :merchant_id, :apiKeyId, :secretApiKey)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        MultiResponse.run(:use_first_response) do |r|
```

Comment:
So MultiResponse is a way to represent more than one request as a single response. By default the status of the last response is proxied as the state of the overall tx. But, if you specify `:user_first_response` instead, it will use the first response as its proxy.

So you want to choose the call that best represents the status of the tx. For a purchase that's really an auth and a capture underneath the covers, you can't say the purchase has succeeded unless the capture has succeeded.

However, for `verify` which is an auth and void, even if the void doesn't succeed the auth response is how we know if the card is valid or not. If the auth succeeds and the void fails, the funds just won't ever be collected, so no harm done. So in that case `:use_first_response` is most accurate.